### ISSUE TITLE
Show returns CIRCULAR when nothing actually circular

### ISSUE THREAD

**Issue by virtulis** [pelotom/runtypes#108]
Running this:

```
import { Record, Partial, Number, String } from 'runtypes';
import show from 'runtypes/lib/show';

const t = Record({
	x: Number,
}).And(Partial({
	y: Number,
}));

console.log(show(t));
```

on 4.0.2 returns:

```
{ x: number; } & { y?: CIRCULAR number; }
```

Doesn't happen when using for example one Number and one String. Works correctly on 4.0.1.

**Comment by virtulis** [comment]
On a side note, how does one even define a circular type with runtypes?

**Comment by pelotom** [comment]
> On a side note, how does one even define a circular type with runtypes?

The `Lazy` operator permits this:

```ts
type Person = { name: string; likes: Person[] };
const Person: Runtype<Person> = Lazy(() => Record({ name: String, likes: Array(Person) }));
```

**Comment by pelotom** [comment]
Should be fixed in v4.0.3, thanks for the report!

**Comment by virtulis** [comment]
Thank you for the project and for the fast fix :) checked the actual use case, all good.
